src/core/Core.js

```diff
@@ -189,12 +189,13 @@ class Uppy {
       this.addThumbnail(newFile.id)
     }
 
-    if (this.opts.autoProceed) {
-      this.upload()
-        .catch((err) => {
+    if (this.opts.autoProceed && !this.scheduledAutoProceed) {
+      this.scheduledAutoProceed = setTimeout(() => {
+        this.scheduledAutoProceed = null
+        this.upload().catch((err) => {
           console.error(err.stack || err.message)
         })
-      // this.bus.emit('core:upload')
+      }, 4)
```

Comment:
This timeout is to debounce uploads being started. Previously this would start a new upload for each individual file that was added, if `autoProceed` is enabled. Eg, dropping 5 files would call upload() 5 times. Since the Transloadit plugin creates an assembly for all files when an upload is started, it's best if it can grab all new files in a single upload() call. With this timeout dropping 5 files calls upload() a single time.